frontend/src/stores/filterState/FilterState.ts

```diff
@@ -36,14 +36,167 @@ export interface FilterLineState {
+      filterLineType,
+      entity
+    );
+
+    if (line) {
+      line.isActive = !line.isActive;
+    }
+  }
 
   /**
-   * States of the filter lines of the edges
+   * Sets the {@link isActive} property of the specified filter line active.
+   * @param filterLineType - filter line which {@link isActive} property is set active
+   * @param entity - specifies whether this entity is a node or an edge
    */
-  nodes: FilterLineState[];
+  public setFilterLineActive(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): void {
+    const line: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (line) {
+      line.isActive = true;
+    }
+  }
+
+  /**
+   * Replaces {@link FilterPropertyState}s of the specified filter line.
+   * @param statesToReplace - the {@link FilterPropertyState}s to be replaced
+   * @param filterLineType - filter line which the {@link FilterPropertyState} is added to
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public replaceFilterPropertyStates(
+    statesToReplace: FilterPropertyState[],
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): void {
+    const lineState: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (lineState) {
+      lineState.propertyFilters = statesToReplace;
+    }
+  }
+
+  /**
+   * Gets {@link FilterPropertyState}s of the specified filter line.
+   * @param filterLineType - filter line which the {@link FilterPropertyState} is added to
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public getFilterPropertyStates(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): FilterPropertyState[] | undefined {
+    const lineState: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    return lineState?.propertyFilters;
+  }
+
+  /**
+   * Gets the values of a {@link FilterPropertyState} in the specified filter line if they exist.
+   * Otherwise undefined is returned.
+   * @param filterLineType - filter line which is searched for
+   * @param filterPropertyName - name of the {@link FilterPropertyState}
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public getPropertyStateValues(
+    filterLineType: string,
+    filterPropertyName: string,
+    entity: 'node' | 'edge'
+  ): string[] | undefined {
+    const propertyState: FilterPropertyState | undefined =
+      this.searchForPropertyState(filterLineType, filterPropertyName, entity);
+
+    return propertyState?.values;
+  }
+
+  private searchForPropertyState(
+    filterLineType: string,
+    filterPropertyName: string,
+    entity: 'node' | 'edge'
+  ) {
+    const line: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (line) {
```

Comment:
Replace all the rest of the function with `line?.find((propertyFilters) => propertyFilters.name === filterPropertyName)`.  
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
This also correctly returns undefined if the value cannot be found.